tests/manage/z_cluster/test_log_rotate.py

```diff
@@ -0,0 +1,190 @@
+                kind=constants.STORAGECLUSTER,
+            )
+            params = '[{"op": "remove", "path": "/spec/logCollector"}]'
+            storagecluster_obj.patch(params=params, format_type="json")
+            log.info(
+                "It takes time for storagecluster to update after the edit command"
+            )
+            time.sleep(30)
+            log.info("Verify storagecluster on Ready state")
+            verify_storage_cluster()
+            ceph_health_check()
+
+        request.addfinalizer(finalizer)
+
+    def get_pod_obj_based_on_id(self, pod_type):
+        """
+        Get Pod Obj
+
+        Args:
+            pod_type (str): The type of pod [osd/mon/mgr/rgw/mds]
+
+        Returns:
+            POD Obj:
+        """
+        pod_objs = self.podtype_id[pod_type][0]()
+        for pod_obj in pod_objs:
+            if self.podtype_id[pod_type][1] == pod.get_ceph_daemon_id(pod_obj):
+                return pod_obj
+
+    def verify_new_log_created(self, pod_type):
+        """
+        Verify new log created on /var/log/ceph
+
+        Args:
+            pod_type (str): The type of pod [osd/mon/mgr/rgw/mds]
+
+        Returns:
+            bool: True if a new log created, otherwise False
+
+        """
+        pod_obj = self.get_pod_obj_based_on_id(pod_type)
+        output_cmd = pod_obj.exec_cmd_on_pod(command="ls -lh /var/log/ceph")
+        expected_string = (
+            self.podtype_id[pod_type][2]
+            if pod_type == "rgw"
+            else f"{self.podtype_id[pod_type][2]}{self.podtype_id[pod_type][1]}"
+        )
+        cnt_logs = len(re.findall(expected_string, output_cmd))
+        if cnt_logs != int(self.podtype_id[pod_type][3]) + 1:
+            log.info(output_cmd)
+            log.error(
+                f"pod_type:{pod_type} cnt_logs_before_fill_log:"
+                f"{self.podtype_id[pod_type][3]} cnt_logs_after_fill_log:{cnt_logs}"
+            )
+            return False
+        return True
+
+    def test_logs_rotate(self):
+        """
+        Test Process:
+            1.Verify the number of MGR,MDS,OSD,MON,RGW logs
+            2.Add logCollector to spec section on Storagecluster
+            3.Write 500M to MGR,MDS,OSD,MON,RGW
+            4.Verify new log created
+            5.Delete logCollector from Storagecluster
+
+        """
+        self.podtype_id = dict()
+        self.podtype_id["mgr"] = [
+            pod.get_mgr_pods,
+            pod.get_ceph_daemon_id(pod.get_mgr_pods()[0]),
+            "ceph-mgr.",
+        ]
+        self.podtype_id["osd"] = [
+            pod.get_osd_pods,
+            pod.get_ceph_daemon_id(pod.get_osd_pods()[0]),
+            "ceph-osd.",
+        ]
+        self.podtype_id["mon"] = [
+            pod.get_mon_pods,
+            pod.get_ceph_daemon_id(pod.get_mon_pods()[0]),
+            "ceph-mon.",
+        ]
+        if config.ENV_DATA["platform"].lower() != constants.AWS_PLATFORM:
```

Comment:
RGW pod will not exist on any cloud provider and here you are checking only aws. This test will fail when run on other cloud platforms. You can use constants.CLOUD_PLATFORMS, so that all the cloud platforms are taken into account.